### PR TITLE
feat: add a way to set `Text` background color

### LINKED ISSUE #69: Add way to set text background color
## What's the motivation for this feature?
It would be nice to set a background color when drawing Text to the screen. This way I can make my text a little more stylish, maybe improve contrast with dark foregrounds, etc.

## Describe the solution you'd like
One solution would be to add another method to Text that also takes `background` parameter (e.g. `Text::write`). Then, the Fill implementation would delegate out to that and choose a default background color.

Example:
```rs
let text = Text::new("Hello world", TextSize::Medium, (0, 0));
text.write(&mut screen, Rgb::WHITE, Rgb::BLACK);
```

## Describe the drawbacks, if any
This would mean you have to use a different method than other widgets to drawing Texts with backgrounds.

## Describe the alternative solutions, if any
Another option would be to add another parameter to the Fill trait.

## Additional context
The implementation should use `vexDisplayBackgroundColor` to set a background color before drawing text.

### PR DESCRIPTION
**Pull request by zabackary**
## Describe the changes this PR makes. Why should it be merged?

This PR adds the `Text::write` API for writing text to a `Display` with a background color.

This PR contains breaking changes. It changes the default background color of text written with the `Fill` trait from the last used background color (which is unpredictable and probably not the best default) to transparent.

## Additional Context

- I have tested these changes on a VEX V5 Brain.
- semver: major (see earlier). If you don't count that as a breaking change, then it's a minor change.
- These changes update the crate's interface (e.g. functions/modules added or changed).
- Fixes #69

### CODE REVIEW

**Review by Gavin-Niederman — DISMISSED** [review]
I would consider moving the background color into the Text itself. This would remove the need for the special `write` syntax. Speaking of `write`, I'm not sure I like that it's called write because it doesn't work like the Write trait where the type implementing Write is being written to. Here the Text is writing to the display.

**Comment by zabackary** [comment]
> I would consider moving the background color into the Text itself. This would remove the need for the special `write` syntax. Speaking of `write`, I'm not sure I like that it's called write because it doesn't work like the Write trait where the type implementing Write is being written to. Here the Text is writing to the display.

I'm going to implement it as discussed [here](https://github.com/vexide/vexide/issues/69#issuecomment-2495151661), if that's okay.

File: CHANGELOG.md [hunk 1]
```diff
@@ -123,6 +124,7 @@ Before releasing:
 - Removed `Position`-to-`Position` `Mul`/`Div` ops, as they were mathematically unsound. Prefer using `Position`-to-scalar operations for this. (#237) (**Breaking Change**)
 - Removed `LinkError::Nul`. (#240) (**Breaking Change**)
 - Removed `LinkError::Port`, because it was broken. VEXlink will no longer perform port validation. (#243) (**Breaking Change**)
+- Added `Fill` implementation for `Text`, since it's a bit of a special case. (#247)
```
Comment (doinkythederp) [inline CHANGELOG.md:127]: This is outdated

File: packages/vexide-devices/src/display.rs [hunk 2]
```diff
@@ -397,6 +409,15 @@ impl Fill for Text {
...
             vexDisplayForegroundColor(color.into().into_raw());
+            vexDisplayBackgroundColor(if let Some(color) = bg_color {
+                color.into_raw()
+            } else {
+                // If the byte before the red component is not 0, VEXos will use
+                // a transparent background.
+                // Not sure this is documented anywhere, but it works and is a
+                // relatively safe assumption that is unlikely to change.
+                0b1 << 24
+            });
```
Comment (doinkythederp) [inline packages/vexide-devices/src/display.rs:421]: This workaround is no longer necessary with the latest changes from main, you can just pass `1` to `vexDisplayPrintf`

File: packages/vexide-devices/src/display.rs [hunk 3]
```diff
@@ -590,10 +611,30 @@ impl Display {
     }
 
     /// Draw a filled object to the display.
+    ///
+    /// For `Text` widgets, the background color is transparent by default. Use
+    /// `Display::fill_text` to specify a background color.
```
Comment (doinkythederp) [inline packages/vexide-devices/src/display.rs:616]: This is outdated because Text doesn't implement Fill

**Review by doinkythederp — CHANGES_REQUESTED** [review]

**Review by doinkythederp — APPROVED** [review]

**Review by Tropix126 — APPROVED** [review]
LGTM

**Pull request merged**